src/components/SampleConfigProvider.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { PropsWithChildren, useContext, useMemo, useState } from "react";
+import { CookiesProvider, useCookies } from 'react-cookie';
+import * as replaceAll from 'string.prototype.replaceall';
+import CodeBlock from '@theme/CodeBlock';
+import Twig from 'twig';
+import { useDebouncedChecker } from "./debouncer";
+
+enum InstanceKind {
+    AUTHZED = 'authzed',
+    SPICEDB = 'spicedb',
+}
+
+type SampleConfigContextValue = {
+    readonly defaultTenant: string,
+    readonly instanceKind: () => InstanceKind | undefined,
+    readonly setInstanceKind: (kind: InstanceKind) => void,
+    readonly token: () => string | undefined;
+    readonly setToken: (token: string) => void;
+    readonly tenant: () => string | undefined;
+    readonly setTenant: (token: string) => void;
+    readonly endpoint: () => string | undefined;
+    readonly setEndpoint: (token: string) => void;
+    readonly buildTempalte: (template: string) => any;
```

Comment:
typo
```suggestion
    readonly buildTemplate: (template: string) => any;
```

Also not sure why the typescript integration with docusaurus didn't complain